src/compiler/java_compiler.nit

```diff
@@ -0,0 +1,2297 @@
+		assert mtype != null
+
+		var nexpr = self.n_expr
+		var nblock = self.n_block
+		if nexpr != null then
+			value = v.expr(nexpr, mtype)
+		else if nblock != null then
+			value = v.new_var(mtype)
+			frame.returnvar = value
+			frame.returnlabel = v.get_name("RET_LABEL")
+			v.add("{frame.returnlabel.as(not null)}: \{")
+			v.stmt(nblock)
+			v.add("\}")
+		else
+			abort
+		end
+
+		v.write_attribute(self.mpropdef.as(not null).mproperty, recv, value)
+		v.frame = old
+		return value
+	end
+end
+
+redef class AExpr
+	# Try to compile self as an expression
+	# Do not call this method directly, use `v.expr` instead
+	private fun expr(v: JavaCompilerVisitor): nullable RuntimeVariable do
+		v.info("NOT YET IMPLEMENTED {class_name}::expr")
+		return null
+	end
+
+	# Try to compile self as a statement
+	# Do not call this method directly, use `v.stmt` instead
+	private fun stmt(v: JavaCompilerVisitor) do expr(v)
+end
+
+redef class ABlockExpr
+	redef fun stmt(v)
+	do
+		for e in self.n_expr do v.stmt(e)
+	end
+	redef fun expr(v)
+	do
+		var last = self.n_expr.last
+		for e in self.n_expr do
+			if e == last then break
+			v.stmt(e)
+		end
+		return v.expr(last, null)
+	end
+end
+
+redef class ASendExpr
+	redef fun expr(v) do
+		var recv = v.expr(n_expr, null)
+		var callsite = callsite.as(not null)
+		var args = v.varargize(callsite.mpropdef, callsite.signaturemap, recv, raw_arguments)
+		return v.compile_callsite(callsite, args)
+	end
+end
+
+redef class ANewExpr
+	redef fun expr(v)
+	do
+		var mtype = self.recvtype
+		assert mtype != null
+
+		if mtype.mclass.name == "NativeArray" then
+			# TODO handle native arrays
+			v.info("NOT YET IMPLEMENTED new NativeArray")
+		end
+
+		var recv = v.init_instance(mtype)
+
+		var callsite = self.callsite
+		if callsite == null then return recv
+
+		var args = v.varargize(callsite.mpropdef, callsite.signaturemap, recv, self.n_args.n_exprs)
+		var res2 = v.compile_callsite(callsite, args)
+		if res2 != null then
+			return res2
+		end
+		return recv
+	end
+end
+
+redef class ASuperExpr
+	redef fun expr(v)
+	do
+		var frame = v.frame
+		assert frame != null
+		var recv = frame.arguments.first
+
+		var callsite = self.callsite
+		if callsite != null then
+			var args
+
+			if self.n_args.n_exprs.is_empty then
+				# Add automatic arguments for the super init call
+				args = [recv]
+				for i in [0..callsite.msignature.arity[ do
+					args.add(frame.arguments[i+1])
+				end
+			else
+				args = v.varargize(callsite.mpropdef, callsite.signaturemap, recv, self.n_args.n_exprs)
+			end
+
+			# Super init call
+			var res = v.compile_callsite(callsite, args)
+			return res
+		end
+
+		var mpropdef = self.mpropdef.as(not null)
+
+		var args
+		if self.n_args.n_exprs.is_empty then
+			args = frame.arguments
+		else
+			args = v.varargize(mpropdef, signaturemap, recv, self.n_args.n_exprs)
+		end
+
+		# Standard call-next-method
+		return v.supercall(mpropdef, recv.mtype.as(MClassType), args)
+	end
+end

```

Comment:
These two comments do not seem to be reported at re right place. Can you precise what was related to that?